### PR TITLE
Video player controller

### PR DESCRIPTION
**Pull request by bmarty**
## Content

Implement a new video player controller, instead of using the one provided by exoplayer.

Note that the module would need some cleanup, in particular some code is in the `api` module and should be moved to the `impl` module.

I prefer to merge this PR first and do the rework after, in a dedicated PR.

## Motivation and context

Be able to render caption, see https://www.figma.com/design/Ni6Ii8YKtmXCKYNE90cC67/Timeline-(new)?node-id=1727-5255&node-type=instance&m=dev.

## Screenshots / GIFs

|Before|After|
|-|-|
|image|image|

## Tests

- Play a video from a room and observe the new controller

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: tests/uitests/src/test/snapshots/images/libraries.designsystem.theme.components_Sliders_Sliders_en.png
Comment (bmarty) [inline tests/uitests/src/test/snapshots/images/libraries.designsystem.theme.components_Sliders_Sliders_en.png:1]: Note: light mode should not be used for now...

File: libraries/dateformatter/api/src/main/kotlin/io/element/android/libraries/dateformatter/api/DurationFormatter.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * - "0:01"
+ * - "0:59"
+ */
+fun Long.toHumanReadableDuration(): String {
+    val inSeconds = this / 1_000
+    val hours = inSeconds / 3_600
+    val minutes = (inSeconds % 3_600) / 60
+    val seconds = inSeconds % 60
+    return if (hours > 0) {
+        String.format(Locale.US, "%d:%02d:%02d", hours, minutes, seconds)
```
Comment (jmartinesp) [inline libraries/dateformatter/api/src/main/kotlin/io/element/android/libraries/dateformatter/api/DurationFormatter.kt:36]: I was going to suggest only calculating the `hours` if `inSeconds > 3600` but it seems like an overkill to just avoid a single allocation and a division 😅 .

File: libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/theme/components/Slider.kt [hunk 2]
```diff
@@ -32,8 +45,20 @@ fun Slider(
     steps: Int = 0,
     onValueChangeFinish: (() -> Unit)? = null,
     colors: SliderColors = SliderDefaults.colors(),
-    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() }
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+    useCustomLayout: Boolean = false,
```
Comment (jmartinesp) [inline libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/theme/components/Slider.kt:49]: This feels a bit weird, to be honest. I think we should either split this component into two (one for the new M3 look and feel and another for the old one) or rename/clarify a bit what `useCustomLayout` does. Maybe rename it to something like `useLegacyMaterialStyle`?

File: libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/theme/components/Slider.kt [hunk 3]
```diff
@@ -43,6 +68,54 @@ fun Slider(
...
+                )
+            }
+        },
+        track = { sliderState ->
+            if (useCustomLayout) {
+                SliderDefaults.Track(
+                    modifier = Modifier.height(8.dp),
+                    colors = colors.copy(
+                        activeTrackColor = Color(0x66E0EDFF),
+                        inactiveTrackColor = Color(0x66E0EDFF),
```
Comment (jmartinesp) [inline libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/theme/components/Slider.kt:104]: Don't we have either semantic or core token colors for these?

File: libraries/mediaviewer/api/src/main/kotlin/io/element/android/libraries/mediaviewer/api/player/MediaPlayerControllerView.kt [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+) {
+    AnimatedVisibility(
+        visible = state.isVisible,
+        modifier = modifier,
+        enter = fadeIn(),
+        exit = fadeOut(),
+    ) {
+        Box(
+            modifier = Modifier
+                .background(color = Color(0x99101317))
```
Comment (jmartinesp) [inline libraries/mediaviewer/api/src/main/kotlin/io/element/android/libraries/mediaviewer/api/player/MediaPlayerControllerView.kt:57]: I guess we also don't have any tokens for this?

**Review by jmartinesp — APPROVED** [review]
It LGTM overall, just a couple of questions about the colors used and a suggestion for the 'legacy' slider component.

File: libraries/mediaviewer/api/src/main/kotlin/io/element/android/libraries/mediaviewer/api/local/LocalMediaViewState.kt [hunk 5]
```diff
@@ -30,7 +30,10 @@ sealed interface PlayableState {
     data object NotPlayable : PlayableState
     data class Playable(
         val isPlaying: Boolean,
-        val isShowingControls: Boolean
+        val progressInMillis: Long,
```
Comment (ganfra) [inline libraries/mediaviewer/api/src/main/kotlin/io/element/android/libraries/mediaviewer/api/local/LocalMediaViewState.kt:33]: This is what we want to expose to the outer world, so I'm not sure this is the right place to add here
I'd have added all those values to the `MediaPlayerControllerState` instead and let the `Playable` derive some values from that

Comment (bmarty) [reply libraries/mediaviewer/api/src/main/kotlin/io/element/android/libraries/mediaviewer/api/local/LocalMediaViewState.kt]: 62134022bbfa3939d3dd28a2aab77b57561f98d7

**Pull request merged**